From: Fabian Vogt <fabian@ritter-vogt.de>
Subject: Fix build with ICU >= 59

ICU >= 59 requires C++11 for its header files.
Qt can't be compiled with -std=c++11 as a whole, so only enable
it for qlocale_icu.cpp.

Index: qt-everywhere-opensource-src-4.8.7/src/corelib/tools/tools.pri
===================================================================
--- qt-everywhere-opensource-src-4.8.7.orig/src/corelib/tools/tools.pri
+++ qt-everywhere-opensource-src-4.8.7/src/corelib/tools/tools.pri
@@ -102,7 +102,15 @@ contains(QT_CONFIG, zlib):include($$PWD/
 else:include($$PWD/../../3rdparty/zlib_dependency.pri)
 
 contains(QT_CONFIG,icu) {
-    SOURCES += tools/qlocale_icu.cpp
+    cpp17.name = cpp17
+    cpp17.input = SOURCES_CPP17
+    cpp17.dependency_type = TYPE_C
+    cpp17.variable_out = OBJECTS
+    cpp17.output = ${QMAKE_VAR_OBJECTS_DIR}${QMAKE_FILE_IN_BASE}$${first(QMAKE_EXT_OBJ)}
+    cpp17.commands = $${QMAKE_CXX} $(CXXFLAGS) -std=c++17 $(INCPATH) -c ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
+    QMAKE_EXTRA_COMPILERS += cpp17
+
+    SOURCES_CPP17 += tools/qlocale_icu.cpp
     DEFINES += QT_USE_ICU
 }
 
--- qt-everywhere-opensource-src-4.8.7/config.tests/unix/icu/icu.pro.orig	2015-05-07 16:14:42.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.7/config.tests/unix/icu/icu.pro	2019-04-18 17:14:05.122419116 +0200
@@ -1,4 +1,11 @@
-SOURCES = icu.cpp
+cpp17.name = cpp17
+cpp17.input = SOURCES_CPP17
+cpp17.dependency_type = TYPE_C
+cpp17.variable_out = OBJECTS
+cpp17.output = ${QMAKE_VAR_OBJECTS_DIR}${QMAKE_FILE_IN_BASE}$${first(QMAKE_EXT_OBJ)}
+cpp17.commands = $${QMAKE_CXX} $(CXXFLAGS) -std=c++17 $(INCPATH) -c ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
+QMAKE_EXTRA_COMPILERS += cpp17
+SOURCES_CPP17 = icu.cpp
 CONFIG -= qt dylib app_bundle
 unix:LIBS += -licuuc -licui18n
 win32:LIBS += -licuin
